recipe_batches/recipe_batches.py

```diff
@@ -3,13 +3,15 @@
 import math
 
 def recipe_batches(recipe, ingredients):
-  batches = 10000000
+  batches = -1
   for k in recipe:
     if k in ingredients.keys():
       x = ingredients[k] // recipe[k]
-      print(x)
-      if x < batches:
-        batches = x
+      if batches != -1:
+        if x < batches:
+          batches = x
+      else:
+          batches = x
     else:
```

Comment:
like the improvement you made, you can also make batches an empty array, and use something like if not all(k in ingredients ... ) something like that